library/core/src/task/wake.rs

```diff
@@ -219,6 +219,15 @@ unsafe impl Sync for Waker {}
 
 impl Waker {
     /// Wake up the task associated with this `Waker`.
+    ///
+    /// As long as the runtime keeps running and the task is not finished, it is
+    /// guaranteed that each invocation of `wake` (or `wake_by_ref`) will be followed
+    /// by at least one `poll` of the task to which this `Waker` belongs. This makes
+    /// it possible to temporarily yield to other tasks while running potentially
+    /// unbounded processing loops.
```

Comment:
This isn't necessarily true; the executor could choose to poll this task again immediately instead of polling other tasks in a fair manner. See related discussion on #74335.

The above guarantee *does* say that you are allowed to yield without breaking your code (just not that it will do anything useful). I'm quite happy to add that to the documentation.

You may want to add the caveat I mentioned to the one you wrote below.